### PR TITLE
Set GIT_DIR correctly if it is not set

### LINKED ISSUE #15744: hook templates have incorrect directory calculation when GIT_DIR is not set
```bash
GIT_DIR=${GIT_DIR:-$(dirname $0)}

for hook in ${GIT_DIR}/hooks/${hookname}.d/*; do
```

if `GIT_DIR` is not set then that first line evaluates to the `hooks` directory containing the script, and thus the 2nd line evaluates to `.../hooks/hooks/...`

the script should read:
```bash
GIT_DIR=${GIT_DIR:-$(dirname $0)/..}

for hook in ${GIT_DIR}/hooks/${hookname}.d/*; do
```

### PR DESCRIPTION
**Pull request by zeripath**
Fix #15744

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

File: modules/repository/hooks.go [hunk 1]
```diff
@@ -22,9 +22,9 @@ import (
...
+		fmt.Sprintf("#!/usr/bin/env %s\ndata=$(cat)\nexitcodes=\"\"\nhookname=$(basename $0)\nGIT_DIR=${GIT_DIR:-$(dirname $0)/..}\n\nfor hook in ${GIT_DIR}/hooks/${hookname}.d/*; do\ntest -x \"${hook}\" && test -f \"${hook}\" || continue\necho \"${data}\" | \"${hook}\"\nexitcodes=\"${exitcodes} $?\"\ndone\n\nfor i in ${exitcodes}; do\n[ ${i} -eq 0 ] || exit ${i}\ndone\n", setting.ScriptType),
+		fmt.Sprintf("#!/usr/bin/env %s\nexitcodes=\"\"\nhookname=$(basename $0)\nGIT_DIR=${GIT_DIR:-$(dirname $0/..)}\n\nfor hook in ${GIT_DIR}/hooks/${hookname}.d/*; do\ntest -x \"${hook}\" && test -f \"${hook}\" || continue\n\"${hook}\" $1 $2 $3\nexitcodes=\"${exitcodes} $?\"\ndone\n\nfor i in ${exitcodes}; do\n[ ${i} -eq 0 ] || exit ${i}\ndone\n", setting.ScriptType),
+		fmt.Sprintf("#!/usr/bin/env %s\ndata=$(cat)\nexitcodes=\"\"\nhookname=$(basename $0)\nGIT_DIR=${GIT_DIR:-$(dirname $0)/..}\n\nfor hook in ${GIT_DIR}/hooks/${hookname}.d/*; do\ntest -x \"${hook}\" && test -f \"${hook}\" || continue\necho \"${data}\" | \"${hook}\"\nexitcodes=\"${exitcodes} $?\"\ndone\n\nfor i in ${exitcodes}; do\n[ ${i} -eq 0 ] || exit ${i}\ndone\n", setting.ScriptType),
```
Comment (silverwind) [inline modules/repository/hooks.go]: Can we use raw strings to make this more readable?

**Review by silverwind — APPROVED** [review]

**Comment by silverwind** [comment]
https://github.com/lithammer/dedent would be even nicer 😉

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
I think the first commit was nice to see the diff itself, and how it is formatted now is totally fine.

to "dedent": I think we should not add to much boilerplate ...

**Comment by silverwind** [comment]
Yes the diff is kind of destroyed with the reformatting. Dedent is optional but I still think worth it because code is much more often read than written and that's just more pleasureable if properly indented :)

**Comment by zeripath** [comment]
> to "dedent": I think we should not add to much boilerplate ...

A go-template or even destroy these hook files and make them unnecessary by using core.hooksPath cleverly.

**Comment by 6543** [comment]
-> #15769

**Pull request merged**